File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(dna1, dna2) {

  if (dna1.length != dna2.length) {
    new Error('DNA strands must be of equal length.');
  }
  else {
    distance = 0;
    for (item = 0; item < dna1.length; item++) {
      if (dna1[item] != dna2[item]) {
        distance += 1;
      }
    }
  }
  
  return distance;

};

module.exports = Hamming;

File: r/hamming/hamming.R
# This is a stub function to take two strings
# and calculate the hamming distance
hamming <- function(strand1,strand2) {
  
  if (nchar(strand1) == nchar(strand2)) {
    sum(unlist(strsplit(strand1, "")) != unlist(strsplit(strand2, "")))
  } else {
    stop("strands need to be of equal length")
  }
  
}
